controllers/operatorconfigmap_controller.go

```diff
@@ -0,0 +1,454 @@
+	_ "embed"
+
+	"github.com/red-hat-storage/ocs-client-operator/pkg/console"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/csi"
+	"github.com/red-hat-storage/ocs-client-operator/pkg/templates"
+
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/apis/csiaddons/v1alpha1"
+	"github.com/go-logr/logr"
+	configv1 "github.com/openshift/api/config/v1"
+	secv1 "github.com/openshift/api/security/v1"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	k8sYAML "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+//go:embed pvc-rules.yaml
+var pvcPrometheusRules string
+
+const (
+	operatorConfigMapName = "ocs-client-operator-config"
+	// ClusterVersionName is the name of the ClusterVersion object in the
+	// openshift cluster.
+	clusterVersionName = "version"
+)
+
+// OperatorConfigMapReconciler reconciles a ClusterVersion object
+type OperatorConfigMapReconciler struct {
+	client.Client
+	OperatorDeployment *appsv1.Deployment
+	OperatorNamespace  string
+	ConsolePort        int32
+	Scheme             *runtime.Scheme
+
+	log               logr.Logger
+	ctx               context.Context
+	consoleDeployment *appsv1.Deployment
+	cephFSDeployment  *appsv1.Deployment
+	cephFSDaemonSet   *appsv1.DaemonSet
+	rbdDeployment     *appsv1.Deployment
+	rbdDaemonSet      *appsv1.DaemonSet
+	scc               *secv1.SecurityContextConstraints
+	operatorConfigMap *corev1.ConfigMap
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (c *OperatorConfigMapReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	clusterVersionPredicates := builder.WithPredicates(
+		predicate.GenerationChangedPredicate{},
+	)
+
+	configMapPredicates := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(client client.Object) bool {
+				namespace := client.GetNamespace()
+				name := client.GetName()
+				return ((namespace == c.OperatorNamespace) && (name == operatorConfigMapName))
+			},
+		),
+	)
+	// Reconcile the ClusterVersion object when the operator config map is updated
+	enqueueConfigMapRequest := handler.EnqueueRequestsFromMapFunc(
+		func(_ context.Context, client client.Object) []reconcile.Request {
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{
+					Name: operatorConfigMapName,
```

Comment:
A config map is  a namespaced resource you need to provide a namespace as well

Reply:
No, as `client.GetNamespace()` will get the client of the watched resource. What you want is the operator namespace. Please look elsewhere in the code to how we obtain it. If not here try OCS operator.